scripts/intersect_tool_yaml.py

```diff
@@ -38,7 +38,15 @@ def main():
             print("Processing: %s" % file)
         a = yaml.safe_load(open(file, 'r'))
         these_tools = a['tools']
+        tools_already_seen_on_this_instance = []
         for tool in these_tools:
+            # deal with tools that have the same name
+            tool_id = tool['name'] + tool['owner']
+            # deal with tools duplicated in different sections
+            if tool_id in tools_already_seen_on_this_instance:
```

Comment:
would you consider just replacing the tools_count key with the `tool_id` you define above? We shouldn't need an extra list for this?

Reply:
Currently `tools_count` doesn't make the distinction between instance.
It's to avoid duplication within instance (not between).

Reply:
sure, but wouldn't the same issue appear between instances as well? Tools with identical names, but different authors, installed on different servers? That would be a false positive we don't want, or?

My reading of the code currently is that if we had 

a/cut and b/cut, which had different functions, the `tools_union` would contain the first seen occurance, but not both, and would mis-count the number of occurrences?

Reply:
### `if tool_id in tools_already_seen_on_this_instance:`

A real example:
```
$ grep -i lcmsmatching -A 4  usegalaxy.*.yml
usegalaxy.eu.yml:- name: lcmsmatching
usegalaxy.eu.yml-  owner: prog
usegalaxy.eu.yml-  revisions:
usegalaxy.eu.yml-  - 253d531a0193
usegalaxy.eu.yml-  tool_panel_section_label: Annotation
--
usegalaxy.eu.yml:- name: lcmsmatching
usegalaxy.eu.yml-  owner: prog
usegalaxy.eu.yml-  revisions:
usegalaxy.eu.yml-  - f86fec07f392
usegalaxy.eu.yml-  tool_panel_section_label: Metabolomics
--
usegalaxy.org.au.yml:- name: lcmsmatching
usegalaxy.org.au.yml-  owner: prog
usegalaxy.org.au.yml-  revisions:
usegalaxy.org.au.yml-  - f86fec07f392
usegalaxy.org.au.yml-  tool_panel_section_label: Metabolomics
```
Without the `if tool_id in tools_already_seen_on_this_instance:` block, the tool is counted 3 times even if it's available only in 2 instances. (note that I'm interesting in tools that are available in the 3 usegalaxy instances)

### `tool_id = tool['name'] + tool['owner']`

A fake example:
```
$ grep -i lcmsmatching -A 4  usegalaxy.*.yml
usegalaxy.eu.yml:- name: cut
usegalaxy.eu.yml-  owner: devteam
--
usegalaxy.eu.yml:- name: cut
usegalaxy.eu.yml-  owner: iuc
--
usegalaxy.org.au.yml:- name: cut
usegalaxy.org.au.yml-  owner: iuc
--
usegalaxy.org.yml:- name: cut
usegalaxy.org.yml-  owner: iuc
```
Without the `tool_id = tool['name'] + tool['owner']`, only the devteam version will be in the output. I didn't investigate deeply and found any case like that in the current tools list but it's just if one day ...

Reply:
Ah with the example it's totally clear what you meant, thanks :)

Reply:
Should we not still update the `tools_count/tools_union` to use the `tool_id`?  To avoid the situation you've described there where there are two cuts, but only one should make the threshold of servers?

Reply:
Damn! A bad copy/paste
It's clearer that it wan't clear for you!

Reply:
Ok, I didn't expect a tool to appear in more than one place.. Nice pick up Gildas! And yes, I agree with Helena.. Let's use the tool_id.